packages/react-router-native/Link.js

```diff
@@ -45,7 +45,7 @@ const __DEV__ = true; // TODO
 if (__DEV__) {
   Link.propTypes = {
     onPress: PropTypes.func,
-    component: PropTypes.func,
+    component: PropTypes.oneOfType([PropTypes.func, PropTypes.instanceOf(React.Component)]),
```

Comment:
```suggestion
    component: PropTypes.elementType,
```
You can just use this instead.